### ISSUE TITLE
Handling of Sets vs Tags

### ISSUE THREAD

**Issue by dettwild** [croneter/PlexKodiConnect#184]
To be honest, I am not even sure if I should label this an 'Issue' or a 'Feature Request', so lets just call it 'Discussion Point' for now ;-).
As you must be painfully aware, Plex currently does not have a proper solution for Movie Sets (especially no  artwork for Sets) and also no separation between such Sets and normal Tags -- both things are currently stuffed into a single Plex field called 'Collections'.
I understand that PKC deals with this shortcoming by feeding the Plex Collection entries to both: Kodi Tags as well as Kodi Sets. 
However, this can cause quite a bit of confusion or even issues depending on the use case the user had in mind. To give you one example: If someone uses the Plex Collections as Tags and one Movie easily can have several such Tags - the Set allocation becomes somewhat meaningless or even random, given that a movie can only be a member of ONE Set. I guess the resulting Set in Kodi is not really 'random' but that the **last** Plex Collection item becomes the Kodi Set, while prior ones probably get assigned as Set, but overwritten later.
Long story short, I think most Plex users either use the Collection field as a Tag or to represent movie Sets, but rarely a combo.
I would propose to add an option to PKC where the user can chose, how to assign his Plex Collections in Kodi. The option would have 3 values:  
- Use as Set only
- Use as Tag only
- Use as Set and Tag 

Note: The current concept of feeding Plex Library names as Tags (and Video Nodes and Playlists) should not be affected by the above proposal as that is a welcome and clever way to allow a users to keep his Kodi organized the same way as Plex.

While this looks like a _**feature request**_, I also came across an effect that I consider a real **_issue_**:
If a movie does NOT have any collection info in Plex, PKC still assigns it to a Set named "" (blank or empty string). This causes display issues if one chooses to show his movies grouped by Set, i.e. all non-set movies showing up in one large pseudo-set.  I think it comes from Plex XML of a movie containing `<Collection.... tag="">` for such items and PKC probably passes this as-is.
If my assumption is right, a solution could be to add a check for empty strings and in such a case NOT to assign anything to Kodi.

Hope that makes sense - looking forward to get feedback on this.

**Comment by dettwild** [comment]
I have found a way to avoid the blank Tags and Sets in Kodi that were mentioned above in the issue part:
In line 987 of **kodidb_functions.py**, replace `self.addTag(kodiid, tag, mediatype)` with
```
            if tag <> '':
            	self.addTag(kodiid, tag, mediatype)
```

and similarly before line 1192 (not counting the above insert yet) add the condition
`            if setname <> '':`

I believe this workaround is by far easier than trying to get Plex to drop empty Collections out of the XML ;-). 
Hope this helps.

**Comment by croneter** [comment]
Thanks for looking into this. I think I found a neater way ;-)

**Comment by dettwild** [comment]
Indeed you did - good stuff.
The commit does not fix the **entire** #184 though like the comment implies ;-)
